From 95cec62006b377634d788d4cca286e46f4dfd54c Mon Sep 17 00:00:00 2001
From: Harini Katakam <harinik@xilinx.com>
Date: Fri, 11 Apr 2014 12:06:28 +0530
Subject: [PATCH] spi: core: Increase timeout value

The existing timeout value in wait_for_completion_timeout is
calculated from the transfer length and speed with tolerance of 10msec.
This is too low because this is used for error conditions such as
hardware hang etc.
The xfer->speed_hz considered may not be the actual speed set
because the best clock divisor is chosen from a limited set such that
the actual speed <= requested speed. This will lead to timeout being
less than actual transfer time.
Considering acceptable latencies, this timeout can be set to a
value double the expected transfer plus 100 msecs.
This patch adds the same in the core.

Signed-off-by: Harini Katakam <harinik@xilinx.com>
Signed-off-by: Mark Brown <broonie@linaro.org>
(cherry picked from commit eee668a92bc775030178c8a7f0d0abe9cce441d7)
(cherry picked from commit e123573ecd0b16f52998e731a0659fdc89d841e4)
---
 drivers/spi/spi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi.c b/drivers/spi/spi.c
index fc36d1d..b280185 100644
--- a/drivers/spi/spi.c
+++ b/drivers/spi/spi.c
@@ -780,7 +780,7 @@ static int spi_transfer_one_message(struct spi_master *master,
 		if (ret > 0) {
 			ret = 0;
 			ms = xfer->len * 8 * 1000 / xfer->speed_hz;
-			ms += 10; /* some tolerance */
+			ms += ms + 100; /* some tolerance */
 
 			ms = wait_for_completion_timeout(&master->xfer_completion,
 							 msecs_to_jiffies(ms));
-- 
1.8.0

